lib/portlayer/storage/volume_cache.go

```diff
@@ -35,22 +36,68 @@ type VolumeLookupCache struct {
 	vlc     map[string]Volume
 	vlcLock sync.RWMutex
 
-	// The underlying data storage implementation
-	volumeStore VolumeStorer
+	// Maps the service url of the volume store to the underlying data storage implementation
+	volumeStores map[url.URL]VolumeStorer
```

Comment:
url.URL has a pointer, are you sure it is a reasonable map key? Have you considered using string representation?